### PR TITLE
NotRetryableError added

### PR DESCRIPTION
**Pull request by yarsky-tgz**

### CODE REVIEW

**Comment by yarsky-tgz** [comment]
I have added test for added error, but coverage check fails.

**Comment by normartin** [comment]
> I have added test for added error, but coverage check fails.

That is really strange. I added `process.exit();` statement to the if branch and the test still passes. We need to fix this first.

File: test/notretryable.error.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {expect} from "./index";
+import {customizeRetry, defaultRetryConfig, notEmpty, retry, wait, NotRetryableError, RetryError} from "../src/retry-promise";
```
Comment (normartin) [inline test/notretryable.error.test.ts]: Can you cleanup unused imports?

File: src/retry-promise.ts [hunk 2]
```diff
@@ -96,6 +96,10 @@ async function _retry<T>(f: () => Promise<T>, config: RetryConfig<T>, done: () =
             }
             config.logger("Until condition not met by " + result);
         } catch (error) {
+            if (error.name === NotRetryableError.name) throw new RetryError(
+              `Met not retryable error. Last error: ${lastError!}`,
+              lastError!
```
Comment (normartin) [inline src/retry-promise.ts]: I think we do not need, not null assertions in both cases.

**Comment by yarsky-tgz** [comment]
> > I have added test for added error, but coverage check fails.
> 
> That is really strange. I added `process.exit();` statement to the if branch and the test still passes. We need to fix this first.

can you show please, where exactly you've added `process.exit();` ?

**Review by normartin — COMMENTED** [review]
> > > I have added test for added error, but coverage check fails.
> > 
> > 
> > That is really strange. I added `process.exit();` statement to the if branch and the test still passes. We need to fix this first.
> 
> can you show please, where exactly you've added `process.exit();` ?



> > > I have added test for added error, but coverage check fails.
> > 
> > 
> > That is really strange. I added `process.exit();` statement to the if branch and the test still passes. We need to fix this first.
> 
> can you show please, where exactly you've added `process.exit();` ?

I tried on my machine. 
```
if (error.name === NotRetryableError.name){
  process.exit();
   throw new RetryError(
              `Met not retryable error. Last error: ${lastError!}`,
              lastError!`);
}
```
This should make the test fail. But it does not. I have no idea why.

File: src/retry-promise.ts (hunk 2 above)
Comment (yarsky-tgz) [reply src/retry-promise.ts]: yeah, and wrong variable was used, please check, hope resolved

**Comment by yarsky-tgz** [comment]
image
image

**Comment by yarsky-tgz** [comment]
```
console.log((new NotRetryableError()).constructor.name)
```

```
Error
```


luv-films-1gEbYgyzAN9Az2oywK

**Comment by yarsky-tgz** [comment]
found solution here:

https://stackoverflow.com/questions/31626231/custom-error-class-in-typescript

**Comment by yarsky-tgz** [comment]
Do you accept idea and implementation as it looks now?

**Comment by normartin** [comment]
Yes, looks good! Will release new version now.

**Pull request merged**